### ISSUE TITLE
was there a problem about wrap_image code?

### ISSUE THREAD

**Issue by GANWANSHUI** [Dawnstar8411/SMAR-Net#1]
class Img_warp_loss(nn.Module):
    def __init__(self):
        super(Img_warp_loss.self).__init__()

    def forward(self, disp, left_img, right_img):
        B, C, H, W = disp.size()

        # mesh grid
        grid = torch.arange(0, W).view(1, -1).repeat(H, 1)
        grid = grid.view(1, 1, H, W).repeat(B, 1, 1, 1)

        vgrid = grid - disp

        vgrid = vgrid.permute(0, 2, 3, 1)
        output = nn.functional.grid_sample(right_img, vgrid)
        mask = torch.ones(right_img.size())
        mask = nn.functional.grid_sample(mask, vgrid)

        mask[mask < 0.999] = 0
        mask[mask > 0] = 1

        img_warp_mask = output * mask
        img_mask = left_img * mask
        img_mask.detach_()
        criterion = torch.nn.L1Loss()
        loss = criterion(img_warp_mask, img_mask)

        return loss, img_warp_mask

when i run the code, error came out that vgrid should be 2 dimensions in the last channels? 
am i miss somethings ?

**Comment by zyl1336110861** [comment]
hello, have you solved the problem? I have encounter the same problem.
